clippy_lints/src/redundant_async_block.rs

```diff
@@ -0,0 +1,51 @@
+use clippy_utils::{diagnostics::span_lint_and_sugg, source::snippet};
+use rustc_ast::ast::*;
+use rustc_errors::Applicability;
+use rustc_lint::{EarlyContext, EarlyLintPass};
+use rustc_session::{declare_lint_pass, declare_tool_lint};
+
+declare_clippy_lint! {
+    /// ### What it does
+    /// Checks for `async` block that only returns `await` on a future.
+    ///
+    /// ### Why is this bad?
+    /// It is shorter to directly use the future.
+    ///
+    /// ### Example
+    /// ```rust
+    /// await {
+    ///     function_returning_a_future().await
+    /// }
+    /// ```
+    /// Use instead:
+    /// ```rust
+    /// function_returning_a_future()
+    /// ```
+    #[clippy::version = "1.69.0"]
+    pub REDUNDANT_ASYNC_BLOCK,
+    complexity,
```

Comment:
Do we have a lintcheck run to ensure we don't get too much churn when introducing this lint?

Nowadays we are quite cautious when adding warn-by-default lints.

Reply:
If you are talking about a `cargo lintcheck`, yes, I ran one (maybe I should mention it in the description). Is there a way to do something similar to a crater run for lint checking?